Source/Workers/decodeDracoPointCloud.js

```diff
@@ -0,0 +1,212 @@
+            if (!defined(attributeType)) {
+                throw new RuntimeError('Error decoding draco point cloud: ' + semantic + ' is not a valid draco semantic');
+            }
+            var attributeId = dracoDecoder.GetAttributeId(dracoPointCloud, attributeType);
+            var attribute = dracoDecoder.GetAttribute(dracoPointCloud, attributeId);
+            var numComponents = attribute.num_components();
+
+            /*eslint-disable no-undef-init*/
+            var quantization = undefined;
+            var transform = new draco.AttributeQuantizationTransform();
+            if (transform.InitFromAttribute(attribute)) {
+                var minValues = new Array(numComponents);
+                for (var j = 0; j < numComponents; ++j) {
+                    minValues[j] = transform.min_value(j);
+                }
+                quantization = {
+                    quantizationBits: transform.quantization_bits(),
+                    minValues: minValues,
+                    range: transform.range()
+                };
+            }
+            draco.destroy(transform);
+
+            transform = new draco.AttributeOctahedronTransform();
+            if (transform.InitFromAttribute(attribute)) {
+                quantization = {
+                    quantizationBits : transform.quantization_bits()
+                };
+            }
+            draco.destroy(transform);
+
+            var vertexArrayLength = numPoints * numComponents;
+            var vertexArray;
+            if (defined(quantization)) {
+                vertexArray = decodeQuantizedDracoTypedArray(dracoPointCloud, attribute, quantization, vertexArrayLength);
+            } else {
+                vertexArray = decodeDracoTypedArray(dracoPointCloud, attribute, vertexArrayLength);
+            }
+
+            results[semantic] = {
+                buffer : vertexArray,
+                quantization : quantization
+            };
+        }
+
+        draco.destroy(dracoPointCloud);
+        return results;
+    }
+
+    function initWorker(dracoModule) {
+        draco = dracoModule;
+        dracoDecoder = new draco.Decoder();
+        self.onmessage = createTaskProcessorWorker(decodeDraco);
+        self.postMessage(true);
+    }
+
+    function decodeDracoPointCloud(event) {
+        var data = event.data;
+
+        // Expect the first message to be to load a web assembly module
+        var wasmConfig = data.webAssemblyConfig;
+        if (defined(wasmConfig)) {
+            // Require and compile WebAssembly module, or use fallback if not supported
+            return require([wasmConfig.modulePath], function(dracoModule) {
+                if (defined(wasmConfig.wasmBinaryFile)) {
+                    if (!defined(dracoModule)) {
+                        dracoModule = self.DracoDecoderModule;
+                    }
+
+                    dracoModule(wasmConfig).then(function (compiledModule) {
+                        initWorker(compiledModule);
```

Comment:
What happens if we are decoding a Draco model and a Draco point cloud at the same time? Since it's a fairly large module, do we want to load it twice?

Reply:
Hm that's a good point.

I guess caching the binary is an option but I'll see if combining `decodeDraco` and `decodeDracoPointCloud` into the same worker is doable. That would also address https://github.com/AnalyticalGraphicsInc/cesium/pull/6559/files/c5870f8cd097b235f67e2e460e7430d8c4506bc4#r185930196.